extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -125,70 +106,35 @@ private async Task<object> CreateTargetObjectAsync(Type targetType, CosmosDBInpu
 
         private async Task<object> CreatePocoAsync(Container container, CosmosDBInputAttribute cosmosAttribute, Type targetType)
         {
-            if (String.IsNullOrEmpty(cosmosAttribute.Id) || String.IsNullOrEmpty(cosmosAttribute.PartitionKey))
+            if (string.IsNullOrEmpty(cosmosAttribute.Id) || string.IsNullOrEmpty(cosmosAttribute.PartitionKey))
             {
                 throw new InvalidOperationException("The 'Id' and 'PartitionKey' properties of a CosmosDB single-item input binding cannot be null or empty.");
             }
 
             ResponseMessage item = await container.ReadItemStreamAsync(cosmosAttribute.Id, new PartitionKey(cosmosAttribute.PartitionKey));
-
-            if (item is null || item.StatusCode is not System.Net.HttpStatusCode.OK || item.Content is null)
-            {
-                throw new InvalidOperationException($"Unable to retrieve document with ID '{cosmosAttribute.Id}' and PartitionKey '{cosmosAttribute.PartitionKey}'");
```

Comment:
By removing this error, the unit tests now fail with a null ref exception instead:

`https://azfunc.visualstudio.com/Azure%20Functions/_build/results?buildId=158152&view=logs&j=1372d9e0-5cd3-5ef5-5e82-7ce7a218d320&t=89898c79-b51f-5488-1a35-13c9114bcb3d&l=218`

```
Expected: Unable to retrieve document with ID '1' a···
Actual:   Object reference not set to an instance o···
```

I think specifically removing the `item.Content is null` part of this check is important - I would suggest we bring back the null check and more intuitive error message.

Reply:
Note: I think these are the only test failing right now, so should be good to go once this is addressed as they are all expecting `Unable to retrieve document with ID '{cosmosAttribute.Id}' and PartitionKey '{cosmosAttribute.PartitionKey}'`:

- ConvertAsync_POCO_ItemResponseNull_ThrowsException_ReturnsFailure
- ConvertAsync_ItemResponse_ResourceIsNull_ThrowsException_ReturnsFailed
- ConvertAsync_POCO_IdProvided_StatusNot200_ThrowsException_ReturnsFailure

Reply:
I'll have to update the unit tests, as this call returning `null` is not how it will behave live.

Reply:
- `ConvertAsync_POCO_ItemResponseNull_ThrowsException_ReturnsFailure` -> removed, we can trust the Cosmos SDK will not return null. If it does, something is really wrong
- `ConvertAsync_ItemResponse_ResourceIsNull_ThrowsException_ReturnsFailed` -> removed, either one of two things is happening: 1) the stream `.Content` is null, which we can trust won't happen - something is wrong with the Cosmos SDK here. Or 2) the stream is not null, but it deserializes to `null`. We should not verify this behavior, it may be a valid (custom converter that returns `null` for example).
- `ConvertAsync_POCO_IdProvided_StatusNot200_ThrowsException_ReturnsFailure` -> updated to be a `CosmosException`. This exception will be more flushed out in a live run, but it validates we at least throw in the non-success cases.